spring-kafka/src/main/java/org/springframework/kafka/support/converter/MessageConverter.java

```diff
@@ -74,4 +76,22 @@ default void commonHeaders(@Nullable Acknowledgment acknowledgment, @Nullable Co
 			.acceptIfNotNull(KafkaHeaders.CONSUMER, consumer, (key, val) -> rawHeaders.put(key, val));
 	}
 
+	@SuppressWarnings("NullAway") // Dataflow analysis limitation
+	default void commonHeaders(@Nullable ShareAcknowledgment acknowledgment, @Nullable ShareConsumer<?, ?> consumer, Map<String, Object> rawHeaders,
```

Comment:
DITTO: `@Nullable Object acknowledgment, @Nullable Object consumer`.
Or might be the case that we don't need this method at all.